Add option to run dns pods on specific nodes

methodx opened the pull request:
Hi

This commit should be handy for those who want to run DNS-service pods on specific types of nodes: master or node.

rhockenbury commented on roles/kubernetes-apps/ansible/templates/kubedns-deploy.yml.j2:
```diff
@@ -33,6 +33,9 @@ spec:
       nodeSelector:
         beta.kubernetes.io/os: linux
+{% if dns_nodetype is defined %}
+        node-role.kubernetes.io/{{ dns_nodetype }}: ""
```
The [node affinity rule ](https://github.com/kubernetes-incubator/kubespray/pull/3746/files#diff-68173b6b58791885bcb383ad8e8f1b85L49) might need to be modified as well.

rhockenbury commented on roles/kubernetes-apps/ansible/templates/coredns-deployment.yml.j2:
```diff
@@ -30,6 +30,9 @@ spec:
       nodeSelector:
         beta.kubernetes.io/os: linux
+{% if dns_nodetype is defined %}
+        node-role.kubernetes.io/{{ dns_nodetype }}: ""
```
Would this conflict at all with the node affinity rule?

Atoms commented:
ci check this

methodx commented:
@rhockenbury Thanks you for the review. I checked everything and it's looks like its better not to set specific node with nodeSelector and use nodeAffinity - in some cases pod will not be able to start at all, if we lose nodes. So I just fixed option for nodeAffinity and renamed my PR.
